src/Range.jsx

```diff
@@ -176,10 +176,17 @@ class Range extends React.Component {
     const { marks, step, min, max } = this.props;
     const cache = this._getPointsCache;
     if (!cache || cache.marks !== marks || cache.step !== step) {
+      const decimal = step.toString().split('.')[1] && step.toString().split('.')[1].length;
       const pointsObject = { ...marks };
       if (step !== null) {
-        for (let point = min; point <= max; point += step) {
+        let point = min;
+        while (point <= max) {
```

Comment:
the performance problem only occur when users set `ste` to be a very small decimals, I think it's not our worries ?